src/serac/physics/heat_transfer.hpp

```diff
@@ -649,34 +655,103 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
                     "boundary condition named \"temperature\".");
     }
 
+    if (is_quasistatic_) {
+      // We store the previous timestep's temperature as the current temperature for use in the lambdas computing the
+      // sensitivities.
+
+      auto [r, drdu] = (*residual_)(differentiate_wrt(temperature_), zero_, shape_displacement_,
+                                    *parameters_[parameter_indices].state...);
+      auto jacobian  = assemble(drdu);
+      auto J_T       = std::unique_ptr<mfem::HypreParMatrix>(jacobian->Transpose());
+
+      for (const auto& bc : bcs_.essentials()) {
+        bc.apply(*J_T, temperature_adjoint_load_vector, adjoint_essential);
+      }
```

Comment:
I had a different question there @jamiebramwell .  What situation would have the essential BCs not be 0 here?

Reply:
This would be non-zero if the quantity of interest was a function of the applied boundary values in the forward problem. This most commonly occurs in inverse scenarios.

Reply:
👍